phys/module_sf_noahdrv.F

```diff
@@ -3740,7 +3740,49 @@ SUBROUTINE lsm_mosaic(DZ8W,QV3D,P8W3D,T3D,TSK,                      &
                      IVGTYP(I,J) == LCZ_6      .or. IVGTYP(I,J) == LCZ_7 .or. IVGTYP(I,J) == LCZ_8 .or. &
                      IVGTYP(I,J) == LCZ_9      .or. IVGTYP(I,J) == LCZ_10 .or. IVGTYP(I,J) == LCZ_11 ) THEN
 
-                  UTYPE_URB = UTYPE_URB2D(I,J) !urban type (low, high or industrial)
+                  !  UTYPE_URB = UTYPE_URB2D(I,J) !urban type (low, high or industrial)
```

Comment:
I still don't understand why this UTYPE_URB and FRC_URB2D are needed here to be re-assigned values. This will cause inconsistency. I remember that in @tslin2 test, he removed this and it worked fine. If this is not removed, then the UTYPE_URB and FRC_URB2D will be different from the input map.

Reply:
This should not happen, as I see the new implementation has been tested. https://github.com/wrf-model/WRF/pull/1986#issuecomment-1918313488

I will give a test when the machine is back

Reply:
@tslin2 @epn09 If mosaic is turned off. Everything worked fine in my machine.

Reply:
It seems mosaic do all the calculations not only for dominant urban land cover grids. So if the grids are not dominant urban types but have some urban areas, there is no information for the urban type in the initial stage for these grids, so the UTYPE_URB2D(i,j) is initialed as zero for these grids.

https://github.com/wrf-model/WRF/blob/a8eb846859cb39d0acfd1d3297ea9992ce66424a/phys/module_sf_noahdrv.F#L3149

https://github.com/wrf-model/WRF/blob/a8eb846859cb39d0acfd1d3297ea9992ce66424a/phys/module_sf_noahdrv.F#L3191


perhaps it's better to use IVGTYP_dominant(I,J) here to only consider mosaic in urban dominant grids?!

or 

do the lsm_mosaic_init first, then urban_var_init in the module_physics_init.F